### ISSUE TITLE
SAML2WebSSOTest-Common

### ISSUE THREAD

**Issue by lina890727** [rmokiem-topdesk/SAML2WebSSOTest-SP#1]
hello.
Download the saml2webssotest IDP and saml2webssotest SP source code. 
You need to refer to saml2webssotest common. 
How to get saml2webssotest common？

**Comment by arucard21** [comment]
The projects use Maven as build system and the SP project defines the common project as dependency. There are several other dependencies as well. So you need to import the projects into an IDE as Maven project. Unfortunately, the SAML2WebSSOTest-Common dependency is not available in any standard Maven repository so it won't be found along with the other dependencies. 

However, an IDE can usually detect the dependency if you have it in the same workspace. So if you have both SAML2WebSSOTest-SP and SAML2WebSSOTest-Common in your IDE, it should detect the common project as dependency. 

If that doesn't work, you can try [installing the common project](http://maven.apache.org/plugins/maven-install-plugin/plugin-info.html) directly to your local machine though I haven't tried this myself. 

Keep in mind, this code is pretty old. It worked the last time I checked but I'm not sure if it still does. But good luck with it.

**Comment by lina890727** [comment]
First of all, thank you very much for your answer .

SAML2WebSSOTest-SP and SAML2WebSSOTest-IDP are downloaded from GitHub by me and uploaded by you. However,SAML2WebSSOTest-SP and SAML2WebSSOTest-IDP are also dependent on SAML2WebSSOTest-Common project.

As you said, under the same IDE, import SAML2WebSSOTest-Common, SAML2WebSSOTest-SP, SAML2WebSSOTest- IDP, then the IDE should detect the public project as a dependency. But the question is, how can I get SAML2WebSSOTest-Common? If you have SAML2WebSSOTest-Common, can you pass it on to me?

**Comment by arucard21** [comment]
You can find the repo here:
https://github.com/arucard21/SAML2WebSSOTest-Common

You might've missed it because the README contains a different name. But this is the correct project.

**Comment by lina890727** [comment]
It has been found. Thank you very much for your help！

**Comment by lina890727** [comment]
After downloading and importing SAML2WebSSOTest-Common,SAML2WebSSOTest-SP can correctly rely on SAML2WebSSOTest-Common. However, SAML2WebSSOTest-IDP is not allowed. It is speculated that the SAML2WebSSOTest-Common code is updated.

So, is there the latest SAML2WebSSOTest-Common？

**Comment by arucard21** [comment]
SAML2WebSSOTest-IDP seems to use version 1.0.7 of SAML2WebSSOTest-Common. So you can just change that in your `pom.xml` file in the SAML2WebSSOTest-IDP project to 1.3.1 (which is the version you have). I'm not entirely sure if that'll work right away, it may not be compatible since a lot would probably have changed in the common project.

Keep in mind that SAML2WebSSOTest-IDP was actually never fully finished and I don't think I ever tested it. So it may not work. I only ever tested SAML2WebSSOTest-SP.

**Comment by lina890727** [comment]
All kinds of methods have been tried. SAML2WebSSOTest-IDP is really not possible

**Comment by arucard21** [comment]
Like I said, that code was never really tested so I'm not sure if it ever worked properly. But when I have some time, I'll look into it and see if I can at least get it in a state where it compiles.

**Comment by arucard21** [comment]
I managed to clean up the code a bit so it at least compiles. I also made some changes to make it a bit easier to use. I moved the 3 repo's to a single repo now. You can find it here:
https://github.com/arucard21/SAML2WebSSOTest

This contains all 3 projects in a single Gradle project (as sub-projects). So you can now clone that repo (or download it) and then import it into your IDE as a Gradle project (instead of Maven). I also updated the code to Java 11 so it should be easy to run on any recent JVM. 

Just from looking at the code again, I think the IdP tests are currently just a copy of the SP tests. So I'm not sure if they do anything. And I already noticed that a lot of it doesn't really work correctly. So right now it's in a state where it at least compiles so you can load it into your IDE. But you may still have to fix some stuff to make it actually work correctly.

**Comment by arucard21** [comment]
Since all the code is now available in a [single repo](https://github.com/arucard21/SAML2WebSSOTest), I'm archiving the other repos. That means this issue will also need to be closed. If you still need help with the code, feel free to open a new issue in the new repo. I'm happy to continue helping with it.
